### ISSUE TITLE
1+1

### ISSUE THREAD

**Issue by joshu9h** [searx/searx#1106]
Gives a internal server error on science tab something todo with wolframalpha when switched off 
it works oke
api?

**Comment by dalf** [comment]
I can reproduce the bug :
```
Traceback (most recent call last):
  File "/home/alexandre/code/searx/ve/lib/python2.7/site-packages/flask/app.py", line 1997, in __call__
    return self.wsgi_app(environ, start_response)
  File "/home/alexandre/code/searx/searx/searx/webapp.py", line 908, in __call__
    return self.app(environ, start_response)
  File "/home/alexandre/code/searx/ve/lib/python2.7/site-packages/werkzeug/contrib/fixers.py", line 152, in __call__
    return self.app(environ, start_response)
  File "/home/alexandre/code/searx/ve/lib/python2.7/site-packages/flask/app.py", line 1985, in wsgi_app
    response = self.handle_exception(e)
  File "/home/alexandre/code/searx/ve/lib/python2.7/site-packages/flask/app.py", line 1540, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/alexandre/code/searx/ve/lib/python2.7/site-packages/flask/app.py", line 1982, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/alexandre/code/searx/ve/lib/python2.7/site-packages/flask/app.py", line 1614, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/alexandre/code/searx/ve/lib/python2.7/site-packages/flask/app.py", line 1517, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/alexandre/code/searx/ve/lib/python2.7/site-packages/flask/app.py", line 1612, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/alexandre/code/searx/ve/lib/python2.7/site-packages/flask/app.py", line 1598, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/alexandre/code/searx/searx/searx/webapp.py", line 512, in index
    result['title'] = highlight_content(escape(result['title'] or u''), search_query.query)
  File "/usr/lib/python2.7/cgi.py", line 1043, in escape
    s = s.replace("&", "&amp;") # Must be done first!
AttributeError: 'float' object has no attribute 'replace'
```

**Comment by asciimoo** [comment]
lol, that's quite a funny bug =]
Seems like the problem is in the json engine which can return float result title. Openaire engine triggers the exception. Working on the fix.

**Comment by dalf** [comment]
Perhaps add some safeguard like in results.py :
```python
            if 'title' in result and not isinstance(result['title'], basestring):
                continue

            if 'content' in result and not isinstance(result['content'], basestring):
                continue

            if 'url' in result and not isinstance(result['url'], basestring):
		continue
```
?

**Comment by asciimoo** [comment]
fixed in 0969e50
